portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DataSourcesList.testcase

```diff
@@ -57,4 +57,52 @@ definition {
 		}
 	}
 
+	@description = "Bug: LRAC-10507 | Automation ID: LRAC-12349 | Test Summary: The data source list has pagination"
+	@priority = "3"
+	test PaginateDataSourceList {
+		var dataSourceList = "";
+
+		TestCase.setUpPortalInstance();
+
+		for (var n : list "1,2,3,4,5,6") {
+			ACSettings.takeTokenFromAnalyticsCloud();
+
+			ACUtils.launchDXP();
+
+			var propertyName = ACDXPSettings.connectDXPtoAnalyticsCloud();
+
+			var dataSource = StringUtil.replace("${propertyName}", " Combined Property", "");
+
+			var dataSourceList = StringUtil.add("${dataSourceList}", "${dataSource}");
+
+			ACDXPSettings.disconnectDXPFromAnalyticsCloud();
+
+			ACUtils.launchAC();
+
+			ACNavigation.goToSettings();
+		}
+
+		task ("Remove the last data source from the list to iterate just the first five") {
+			var newDataSourceList = ListUtil.newListFromString("${dataSourceList}");
+
+			ListUtil.remove("${newDataSourceList}", "${dataSource}");
+		}
+
+		task ("Change data source pagination to 5 items per page and assert the data sources on the page") {
+			ACUtils.changePagination(itemsPerPage = "5");
+
+			for (var dataSourceName : list "${newDataSourceList}") {
+				ACUtils.assertDataSourceName(dataSourceName = "${dataSourceName}");
+			}
+
+			ACUtils.assertItemsNameNotPresent(itemList = "${dataSource}");
+		}
+
+		task ("Go to the second page and assert the last data source") {
+			ACUtils.checkAnyPage(pageNumber = "2");
+
+			ACUtils.assertDataSourceName(dataSourceName = "${dataSource}");
+		}
+	}
+
```

Comment:
Just wanna make sure I understand the logic.

- So you're taking in the list of data sources as they're being created with ${dataSourceList}. 
- You use ${dataSource} to retrieve the last data source created. 
- Using ${newDataSourceList} you remove the last created data source ${dataSource}. 
- Then changing pagination check that ${dataSource} should not be in the list
- Then changing page to check it would be in the list

For the last step should we also check that ${newDataSourceList} is not preset?

Reply:
Looks good though! Let me know if I got anything wrong

Reply:
Yes, you got it right! I was thinking about checking the ${newDataSourceList} is not present too but I don't know if it would be too much, what do you think Josh?

Reply:
I agree with Josh, I think it would be good to check that the first 5 data sources don't appear on the second page.

I think you should also check the total data source that appears in the pagination of each page. At the end of the table is the information (Showing 1 to 15 of 15 entries.). I think this was the bug that was occurring.

You can use:
`ACUtils.viewResultsMessage(results = "Showing 1 to 5 of 7 entries.");`

Reply:
Vini has a good point, the bug was happening with the result message so the purpose of this test should also include the result message.

Reply:
Thanks guys!